dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/support/AbortPolicyWithReport.java

```diff
@@ -78,43 +78,34 @@ private void dumpJStack() {
             return;
         }
 
-        Executors.newSingleThreadExecutor().execute(new Runnable() {
-            @Override
-            public void run() {
-                String dumpPath = url.getParameter(Constants.DUMP_DIRECTORY, System.getProperty("user.home"));
-
-                SimpleDateFormat sdf;
-
-                String os = System.getProperty("os.name").toLowerCase();
-
-                // window system don't support ":" in file name
-                if(os.contains("win")){
-                    sdf = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss");
-                }else {
-                    sdf = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss");
-                }
-
-                String dateStr = sdf.format(new Date());
-                FileOutputStream jstackStream = null;
-                try {
-                    jstackStream = new FileOutputStream(new File(dumpPath, "Dubbo_JStack.log" + "." + dateStr));
-                    JVMUtil.jstack(jstackStream);
-                } catch (Throwable t) {
-                    logger.error("dump jstack error", t);
-                } finally {
-                    guard.release();
-                    if (jstackStream != null) {
-                        try {
-                            jstackStream.flush();
-                            jstackStream.close();
-                        } catch (IOException e) {
-                        }
-                    }
-                }
-
-                lastPrintTime = System.currentTimeMillis();
+        ExecutorService pool = Executors.newSingleThreadExecutor();
+        pool.execute(() -> {
+            String dumpPath = url.getParameter(Constants.DUMP_DIRECTORY, System.getProperty("user.home"));
+
+            SimpleDateFormat sdf;
+
+            String os = System.getProperty("os.name").toLowerCase();
+
+            // window system don't support ":" in file name
+            if (os.contains("win")) {
+                sdf = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss");
+            } else {
+                sdf = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss");
+            }
+
+            String dateStr = sdf.format(new Date());
+            //try-with-resources
+            try (FileOutputStream jStackStream = new FileOutputStream(new File(dumpPath, "Dubbo_JStack.log" + "." + dateStr))) {

```

Comment:
Here you should need to empty the buffer area , do u think? `jstackStream.flush();`

Reply:
I don't think it's necessary to use the flush method for FileOutputStream. Caching is not used in FileOutputStream, nor does FileOutputStream override the flush method of the OutputStream parent. Even if called, it is an empty method of the parent class.do u thind so?